### PR TITLE
feat(search): add filters to replace removed document search

### PR DESCRIPTION
**Pull request by Yelinz**
Add tag and category filter to file filters.
Filter for document title and description on fulltextsearch filter.

Drive by:
close https://github.com/projectcaluma/alexandria/pull/659
Remove category filter on document in favor for categories

### CODE REVIEW

File: alexandria/core/filters.py [hunk 1]
```diff
@@ -243,7 +253,11 @@ def search_files(self, queryset, name, value):
         queryset = queryset.annotate(
             search_rank=SearchRank(F("content_vector"), search_query),
             search_context=SearchHeadline(F("content_text"), search_query),
-        ).filter(content_vector=search_query)
+        ).filter(
+            Q(content_vector=search_query)
+            | Q(document__title__icontains=value)
+            | Q(document__description__icontains=value)
```
[the remark this answers is not in the document]
Comment (Yelinz) [reply alexandria/core/filters.py:260]: Would mean updating the title and description needing to also update the `content_vector`.
I would leave them separate from the `content_text` and try to work with the weights as we have 4 weight options and 4 different pieces of information.

[the remark this answers is not in the document]
Comment (winged) [reply alexandria/core/filters.py:260]: That would imply not being able to propperly calculate the search context anymore (see the `SearchHeadline` expression a few lines further up), hence the suggestion.

If you can figure out how to do a proper `SearchHeadline` with the fields separate as you suggest, then yeah I think that'd work fine (I think you should be able to pass in a concat expression into `SearchHeadline`)

File: alexandria/core/filters.py [hunk 2]
```diff
@@ -114,7 +123,7 @@ def filter(self, qs, value):
             synonyms = models.Tag.objects.filter(
                 Q(id=tag) | Q(tag_synonym_group__tags__id=tag)
             )
-            qs = qs.filter(tags__in=synonyms)
+            qs = qs.filter(Q(**{f"{self.field_name}__in": synonyms}))
```
Comment (winged) [inline alexandria/core/filters.py:126]: No need for the `Q`

```suggestion
            qs = qs.filter(**{f"{self.field_name}__in": synonyms})
```

**Review by winged — APPROVED** [review]
Nice work, I like it! Just a minor nitpick (feel free to ignore or fix at your discretion, it's not too important)

File: alexandria/core/filters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yelinz) [reply alexandria/core/filters.py]: Done with `Concat`

File: alexandria/core/filters.py (hunk 2 above)
Comment (Yelinz) [reply alexandria/core/filters.py]: Applied in the correct commit

**Pull request merged**